sysid-application/src/main/native/include/sysid/view/Generator.h

```diff
@@ -20,31 +20,55 @@
 
 #include "sysid/deploy/DeploySession.h"
 #include "sysid/generation/ConfigManager.h"
+#include "sysid/generation/HardwareType.h"
 
 namespace sysid {
+/**
+ * Constexpr way of storing a const char* array of HardwareType names to be used
+ * with ImGui's const char* setup. Adapted from:
+ * https://stackoverflow.com/a/34465458
+ *
+ * @tparam S The size of the array storing the HardwareTypes that will be used.
+ */
+template <size_t S>
+struct DisplayNameStorage {
+  /**
+   * A const char* array to hold all the HardwareType names
+   */
+  const char* names[S];
+
+  /**
+   * An int to pass into the size parameter of functions like `ImGui::Combo`
+   */
+  const int size{S};
```

Comment:
Could you add a blank line between the member variable and the constructor definition?